core/src/main/java/io/grpc/GrpcSession.java

```diff
@@ -0,0 +1,35 @@
+package io.grpc;
+
+import java.net.SocketAddress;
+import java.security.cert.Certificate;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+
+/**
+ * Holds state pertaining to a single transport connection.
+ */
+public class GrpcSession {
```

Comment:
Having a single reference identifying the TCP connection could be useful, but it would also generally be abused. I wonder if we can get away, long-term, with not exposing which RPCs are on the same TCP connection. If we're able to, that would be preferable. If we are going to have to expose it anyway, then we might as well now. I'm going to have to mull over that.